SwrveSDK/src/main/java/com/swrve/sdk/conversations/ui/video/WebVideoViewBase.java

```diff
@@ -1,43 +1,44 @@
-package com.swrve.sdk.conversations.ui;
+package com.swrve.sdk.conversations.ui.video;
 
 import android.content.Context;
 import android.content.Intent;
 import android.net.Uri;
 import android.util.Log;
-import android.view.MotionEvent;
 import android.view.View;
 import android.view.ViewGroup;
 import android.webkit.WebChromeClient;
 import android.webkit.WebView;
 import android.webkit.WebViewClient;
-import android.widget.FrameLayout;
 import android.widget.Toast;
 
 import com.swrve.sdk.SwrveHelper;
 import com.swrve.sdk.conversations.engine.model.Content;
 import com.swrve.sdk.conversations.engine.model.ConversationAtom;
-
-public class HtmlVideoView extends WebView implements ConversationContent {
-    private static final String LOG_TAG = "SwrveSDK";
-    public static final String PLAYER_VIDEO_VIMEO = "vimeo";
-    public static final String PLAYER_VIDEO_YOUTUBE = "youtube";
-    private static final String EXTERNAL_MARKER = "&swrveexternal";
-    private String url;
-    private int height;
+import com.swrve.sdk.conversations.ui.ConversationContent;
+import com.swrve.sdk.conversations.ui.ConversationFullScreenVideoFrame;
+
+public abstract class WebVideoViewBase extends WebView implements ConversationContent {
+    protected static final String LOG_TAG = "SwrveSDK";
+    protected static final String PLAYER_VIDEO_VIMEO = "vimeo";
+    protected static final String PLAYER_VIDEO_YOUTUBE = "youtube";
+    protected static final String EXTERNAL_MARKER = "&swrveexternal";
+    protected String url;
+    protected int height;
     // This is a link to the video to be displayed in case the video is not visible to the customers
-    private String errorHtml;
+    protected String errorHtml;
     // This is the final output. This will be the tags and information that is rendered in the application
-    private String videoHtml;
-    private Content model;
-    private ConversationFullScreenVideoFrame fullScreenContainer;
+    protected String videoHtml;
+    protected Content model;
+    protected Context context;
+    protected ConversationFullScreenVideoFrame fullScreenContainer;
+    protected int width;
+
 
-    public HtmlVideoView(Context context, Content model, ConversationFullScreenVideoFrame fullScreenContainer) {
+    public WebVideoViewBase(Context context, Content model, ConversationFullScreenVideoFrame fullScreenContainer) {
         super(context);
+        this.context = context;
```

Comment:
I could use getContext() each time I need it?

Its used for getting assets in the subclass view